### DISCUSSION TITLE
Game mode suggestions

### DISCUSSION THREAD

**Discussion by finnbear** [SoftbearStudios/mk48#163] [Ideas]
Ideas:
- Arcade/casual vs. realistic/pro
- FFA vs. teams (suggested by 8151382425 among others)
- Freeplay vs. Conquest (suggested by Cris)
- Large 50/50 team battle (suggested by Admiral)
- No points required to upgrade (Hang)
- Domination: start as Skipjack and capture majority of 5 points (Suggested by Type)
- Capture the flag (suggested by SS NERO)
- Base wars (suggested by SS NERO)
- Peaceful/sandbox mode (suggested by Jake)
- Classic, WW2, Modern etc. (suggested by Capt Grey, Winter)
- Pearl Harbor / Carriers vs. Battleships (suggested by Awesome)
- 1v1 (suggested by Camo, Crusader)
- Impostor Mode (suggested by IcyBun)
- Zubr Race (suggested by Camo)
- Low-tier only (suggested by ꧁🦈Ҡinͥgfiͣsͫh≛꧂#9975)
- Zombie (smol bot swarm) mode (suggested by russianspy81)
- Zombie (Olympias edition) (suggested by jeffacce)
- Battleship mode (no missiles) (suggested by Winter)
- Base attack (players vs. base with limited upgrades) (suggested by Lighter)

If and when it comes time to add game modes, we will pick/combine ideas from this list.

**Comment by caibear** [comment]
If we have an FFA mode utility ships such as the dredger become obsolete because you can't have your team defend you.

**Comment by criscola** [comment]
# Conquest:

The basic idea would be 2 teams fighting against each other. The map should contain a few flags (could be also called objectives) proportional to its size , e.g. 3 flags. Each flag has an effect radius in which ships can go and start to capture it. The team which holds more flags the longest period of time wins. 

## Details:

- The flag needs a certain amount of time to be captured.
- More ships in the effect radius = the flag is captured faster.
- If ships from both opponent teams are in the radius area, the capture is slowed down. If an equal number of ships from both teams are in the effect radius, the capture is paused, you should get the gist.
- Teams start from their main base, where the opponent players cannot go. Each player can spawn at an objective captured by its team or in their main base, but not if an objective is contested (this point to be elaborated).

## Win condition:

Here we could have some variations:
1. There is a counter per team which is decremented by the opponent team. The rate of decrease is calculated proportionally to the number of flags held by the opponent team. E.g., if one team holds 3/3 flags, the other team will see its counter decrease fast. If it holds 2/3, the counter decrease fast but not that fast, if it holds 1/3, the counter is decreased slowly
2. There is a counter per team which is incremented by said team. The rate of increase is calculated proportionally to the number of flags held by the team. When the counter reaches a certain number, the game is finished. The team completing their counter the quickest, wins.
3. 1 or 2, but with a time limit, so the fight doesn't drag on too much if the teams keep staying even for too long.

We could even add more stuff to balance, e.g. if the two teams are severly unbalanced, the losing team could get a big battleship to try to retake ground as some kind of reinforcement.

**Comment by caibear** [comment]
Regarding @criscola's idea: Instead of a time limit, the speed the counters move could increase over time.

**Comment by thatretrodev** [comment]
That could be implemented in theory by having the server switch game modes randomly every couple minutes to every couple hours or by having multiple servers for each game mode.

**Comment by caibear** [comment]
> That could be implemented in theory by having the server switch game modes randomly every couple minutes to every couple hours or by having multiple servers for each game mode.

I most like the idea of game modes being naturally built into certain locations in the world rather than having multiple servers or shuffling game modes. This would keep all players in the same place and could increase the sense of community. Although this would require a more thoughtful implementation.

**Comment by IcyBun** [comment]
Maybe a gamemode where alot of different player are controlling a big ship [ level 4 or above ], every each of them control different thing, maybe weaponry or movement, and doing maybe a 1v1 or 5v5 team death match, pretty sure that's quite possible, but other ideas are better too

**Comment by Titan2931** [comment]
I like the idea of an offline sandbox mode
